### ISSUE TITLE
Memory Leak

### ISSUE THREAD

**Issue by mikihiro-t** [HiraokaHyperTools/PDFsharp#7]
Thank you for your kenjiuno.PdfSharp.
I have made a solution with this.

Convert FlowDocument to XPS.  And then Convert XPS to PDF files. 

Problem is when many XPS Files to PDF files, this application crashes.

For Example,

```C#
      for (int i = 0; i < 500; i++)
      {
        var xpsPath = @"C:\test\test.xps";
        PdfSharp.Xps.XpsConverter.Convert(xpsPath);
      }
```
Process Memory increases:
![image](https://user-images.githubusercontent.com/29480969/58005108-e7858280-7b1f-11e9-9ec7-40f873ad60ea.png)

SnapShot of Memory:
Memory sizes with PdfSharp.Fonts are increased.
![image](https://user-images.githubusercontent.com/29480969/58005156-fe2bd980-7b1f-11e9-88ab-a014f871b670.png)

test.xps is like this
![image](https://user-images.githubusercontent.com/29480969/58005300-49de8300-7b20-11e9-8beb-9aa70996716e.png)

Could you help me?

Note:
PdfSharp.Drawing\XFontFamily.cs
I have changed this line(267) for Visual Studio 2017
old
```C#
    families = FontFamily.GetFamilies(graphics.gfx);
```
new
```C#
    families = FontFamily.Families;
```

**Comment by kenjiuno** [comment]
@mikihiro-t thanks for report! I'll check about this

**Comment by kenjiuno** [comment]
Ok I have fixed in #8
1.1.3 will be available soon
https://www.nuget.org/packages/kenjiuno.PdfSharp.Xps/1.1.3

**Comment by mikihiro-t** [comment]
Thank you for your help(1.1.3). I will try.

I have traced the code(1.1.2).
This static Global has not disposed.
When a xps file converted, Global.table.Count increses.

 I'm not sure this code is good. But this works.

**FontDescriptorStock.cs**
![image](https://user-images.githubusercontent.com/29480969/58062000-9030f200-7bb3-11e9-9c19-7023276f3324.png)
```C#
    public static void DisposeGlobal()
    {
      global = null;
    }
```
**XpsConverter.cs**

At the last in Convert Method ,  call DisposeGlobal.

**public static void Convert(string xpsFilename, string pdfFilename, int docIndex, bool createComparisonDocument)**

![image](https://user-images.githubusercontent.com/29480969/58062079-f1f15c00-7bb3-11e9-878d-00a98ff16fb9.png)
```C#
PdfSharp.Fonts.FontDescriptorStock.DisposeGlobal(); 
```
Maybe this **method(Stream)** needs the same code.

**public static void Convert(Stream xpsInStream, Stream pdfOutStream, bool closePdfStream)**

![image](https://user-images.githubusercontent.com/29480969/58062168-4bf22180-7bb4-11e9-9de1-827929ef8a36.png)

**Comment by mikihiro-t** [comment]
Thank you very much. It was very helpful.
Ver 1.1.3 works well.

Note:  FontFamily.GetFamilies(graphics.gfx) 
turning off "Warning as error"
https://forum.pdfsharp.net/viewtopic.php?f=2&t=903
So I can use the original code.

**Comment by kenjiuno** [comment]
> ```cs
> PdfSharp.Fonts.FontDescriptorStock.DisposeGlobal(); 
> ```

I'm not sure that keep using `FontDescriptorStock.global` is good idea.

So, in 1.1.3 I have added `FontDescriptorStock.NewInstance` and replaced all `Global` references to `NewInstance` not to rely on global cache mechanism.

I'm not sure about this too, however I have selected this since:
- both versions (using Global or NewInstance) generates almost same size PDF files.
- embedded tests succeed.

> PdfSharp.Drawing\XFontFamily.cs
> I have changed this line(267) for Visual Studio 2017

Thanks, I have fixed it in https://github.com/HiraokaHyperTools/PDFsharp/pull/6
Current `master` branch should provide merged code.
